clippy_lints/src/methods/mod.rs

```diff
@@ -3632,6 +3633,36 @@ declare_clippy_lint! {
     "`as_str` used to call a method on `str` that is also available on `String`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Looks for one of two kinds of expressions:
+    /// - Calling `<T as TryInto<U>>::try_into` where `T` also implements `Into<U>`
+    /// - Calling `<T as TryFrom<U>>::try_from` where `T` also implements `From<U>`
+    ///
+    /// ### Why is this bad?
+    /// In those cases, the `TryInto` and `TryFrom` trait implementation is a blanket impl which forwards
+    /// to `Into` or `From`, which always succeeds.
+    /// Indeed, its associated `Error` type is the empty enum `Infallible`,
+    /// which is impossible to construct by nature, so the error handling that is then imposed by
+    /// the `Result<_, Infallible>` is unnecessary and can be avoided simply by calling `.into()` or `::from`
+    /// directly in the first place.
```

Comment:
This also has more detail than is necessary to get the point across. The key point being the need to unwrap the result even though it's always going to be the `Ok` variant.

Reply:
Changed the documentation a bit. Is this better?